### PR TITLE
Ems

### PR DESCRIPTION
**Pull request by EMSwank**
Almost done with the encrypt logic

### CODE REVIEW

File: lib/dictionary.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                   "0"=> 27,
+                   "1"=> 28,
+                   "2"=> 29,
+                   "3"=> 30,
+                   "4"=> 31,
+                   "5"=> 32,
+                   "6"=> 33,
+                   "7"=> 34,
+                   "8"=> 35,
+                   "9"=> 36,
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/dictionary.rb]: Sticking with quotes because that is the preferred way at Turing.

File: lib/enigma.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+require 'Date'
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/enigma.rb]: Not clear what this does, so we are erring on the side of caution and excluding for now.

File: lib/enigma.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+require 'Time'
+require 'pry'
+require './lib/dictionary'
+
+class Enigma
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/enigma.rb]: Trying not to include comments that are not 100% necessary

File: lib/enigma.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  def key_offset(key = random_key(5, 10))             #problematic for edge cases? can we hard code in random_key and still have appropriate tests
+    key.map.with_index do |element , index|
+      "#{element}#{key[index + 1]}".to_i
+    end.shift(4)
+  end
+
+  def date_string
+    Date.today.strftime("%m%d%y")
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/enigma.rb]: Sticking with double quotes.

File: lib/enigma.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  def date_offset
+    date_squared = date_string.to_i ** 2
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/enigma.rb]: Keeping for stylistic reasons/readability

File: lib/enigma.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    date_squared = date_squared.to_s.split("")
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/enigma.rb]: Sticking with double quotes.

File: test/enigma_test.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  def test_it_can_combine_arrays
+    enigma = Enigma.new
+
+    assert_equal [6, 8, 10, 12], enigma.master_key([4,5,6,7], [2,3,4,5])
+  end
+
+  def test_it_can_translate_characters_to_correct_position
+    enigma = Enigma.new
+    expected = [[8, 5, 12, 12], [15, 37, 23, 15], [18, 12, 4]]
+    message = "hello world"
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply test/enigma_test.rb]: Sticking with double quotes.
